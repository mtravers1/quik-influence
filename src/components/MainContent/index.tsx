import { Stack, Flex, Divider, Box, useColorMode } from '@chakra-ui/react';
import { css } from '@emotion/react';
import Header from 'components/Header';
import SideBarMenu from 'components/SideBarMenu';
import {
  bgThemeColor,
  themeColor,
  dashboardColor,
} from 'utils/constants/colorConstants';
import useResponsiveFilter from 'hooks/useResponsiveFilter';
import { AccountUpdatePrompt } from 'components/AccountUpdatePromt';

interface MainContentProps {
  children: React.ReactElement;
  filter?: React.ReactElement;
}

const MainContent = ({ children, filter }: MainContentProps) => {
  const { colorMode } = useColorMode();
  const { open, openFilter, responsiveFilterMode, toggleFilter } =
    useResponsiveFilter();

  const filterOpenStyles = css`
    & {
      position: fixed;
      height: 100%;
      background: ${dashboardColor[colorMode]};
      right: ${openFilter ? '-15px' : '-700px'};
      top: 61px;
      padding: 20px;
      border-left: 1px solid #000;
      transition: 0.3s ease;
    }
  `;

  return (
    <>
      <Stack position="relative">
        <Header
          bgColor={bgThemeColor[colorMode]}
          color={themeColor[colorMode]}
          toggleFilter={toggleFilter}
          showFilter={!!filter && responsiveFilterMode}
        />
        <Flex
          flexDirection="row"
          position="relative"
          css={css`
            & {
              margin-top: 1px !important;
            }
          `}
        >
          <SideBarMenu
            bgColor={bgThemeColor[colorMode]}
            colorMode={colorMode}
            open={open}
          />
          <Divider
            bgColor={bgThemeColor[colorMode]}
            orientation="vertical"
            height="100vh"
          />
          <Flex
            flexGrow={1}
            bgColor={dashboardColor[colorMode]}
            color={themeColor[colorMode]}
            px={20}
            py={10}
            overflow="hidden"
          >
            <Box flexGrow={1} width="100%" mx="auto" overflowX="scroll">
              {children}
            </Box>
            {filter && (
              <Box
                flexShrink={0}
                marginX="15px"
                css={
                  responsiveFilterMode
                    ? filterOpenStyles
                    : css`
                        & {
                          position: sticky;
                          top: 0;
                        }
                      `
                }
              >
                {filter}
              </Box>
            )}
          </Flex>
        </Flex>
      </Stack>
      <AccountUpdatePrompt />
    </>
  );
};

export default MainContent;
