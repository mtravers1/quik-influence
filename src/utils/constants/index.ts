export const CONTENT_URL = '/content';
export const Q_TOKEN = '_q_inf';
export const DEFAULT_PAGE_SIZE = '100';
export const ADMINS_ID = [2];

export const FILE_STATUS = {
  PENDING: 'pending',
  PROCESSING: 'processing',
  UPLOADING: 'uploading',
  PAUSED: 'paused',
  COMPLETED: 'completed',
  FAILED: 'failed',
  IDLE: 'idle',
};

export const headers: string[] = [
  'phone',
  'avatar',
  'email',
  'firstName',
  'lastName',
  'address',
  'city',
  'dateOfBirth',
  'instagramId',
  'socialMediaHandle',
  'state',
  'country',
  'postalCode',
  'gender',
  'address2',
  'address3',
  'facebookHandle',
  'tiktokHandle',
  'twitterHandle',
];

export const formDataRelatedToSpecificUser = [
  'updatedAt',
  'twitterHandle',
  'avatar',
  'phone',
  'address2',
  'email',
  'address1',
  'postalCode',
  'socialMediaHandle',
  'lastName',
  'firstName',
  'instagramId',
  'tiktokHandle',
  'address',
  'address3',
  'facebookHandle',
];

export const MARKETING_ADMIN = ['can_approve_or_reject_campaign'];
export const ADMIN_USER_AFFILIATE = ['can_get_joinable_campaigns'];
export const CAN_HANDLE_INFLUENCERS = ['can_manage_influencers'];

export const MARKET_PLACE_NEW_PRODUCTS = 'new';
export const MARKET_PLACE_POPULAR_PRODUCTS = 'popular';

export const CART_CLICK_NAME = 'cart';
