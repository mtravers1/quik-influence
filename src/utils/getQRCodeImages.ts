import QRCode, { QRCodeToDataURLOptions } from 'qrcode';
import { QrCodeRawData } from 'types';

const defaultOptions: Partial<QRCodeToDataURLOptions> = {
  errorCorrectionLevel: 'M',
  width: 256,
};

export default async function getQRImages(
  list: QrCodeRawData[],
  options?: QRCodeToDataURLOptions,
): Promise<string[]> {
  const qrCodeUrlPromises = list.map(({ text }) =>
    QRCode.toDataURL(text, { ...defaultOptions, ...options }),
  );
  const images = await Promise.all(qrCodeUrlPromises);
  return images;
}
