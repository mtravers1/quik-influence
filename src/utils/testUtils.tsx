import { FC, ReactElement } from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from 'redux/reducers';

function makeTestStore(rootReducer: any, initialState: any) {
  const store = createStore(rootReducer, initialState, applyMiddleware(thunk));
  const origDispatch = store.dispatch;
  store.dispatch = jest.fn(origDispatch);
  return store;
}

const renderWithStore = (ui: ReactElement, storedefaults: any) => {
  const { initialState, ...renderOptions } = storedefaults;
  const store = makeTestStore(rootReducer, initialState);

  const Wrapper: FC = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
  };

  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

export * from '@testing-library/react';
export { renderWithStore };
