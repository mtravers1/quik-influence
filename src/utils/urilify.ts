export function urlify(text: string) {
  var urlRegex = /(https?:\/\/[^\s]+)/g;
  return text.replace(urlRegex, function (url) {
    return '<a style="color: #007FFF;" href="' + url + '">' + url + '</a>';
  });
}
