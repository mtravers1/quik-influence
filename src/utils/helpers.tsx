import axios from 'axios';

// const baseurl = 'http://localhost:2022';
const baseurl = process.env.NEXT_PUBLIC_BACKEND_URL;

import { DropdownSelectOption } from 'components/DropdownSelect';

const _axiosInstance = axios.create({
  baseURL: `${baseurl}/api/v1`,
  headers: {
    'Access-Control-Allow-Headers':
      'Access-Control-Allow-Origin, Access-Control-Allow-Headers, Content-Type',
    'Access-Control-Allow-Origin': '*',
  },
});

_axiosInstance.interceptors.request.use((config: any) => {
  let token: any = '';
  if (typeof window !== 'undefined') {
    token = localStorage.getItem('token');
  }
  config.headers.token = token;
  return config;
});

export const axiosInstance = _axiosInstance;

export const validate = (field: any, Regex: any, pattern: any) => {
  if (pattern.test(field)) return true;
  return false;
};

export const setToken = (token: string) => {
  axiosInstance.interceptors.request.use((config: any) => {
    config.headers.token = token ? token : '';
    return config;
  });
};

export const getNumberRange = (
  start: number,
  stop: number,
  step: number
): DropdownSelectOption[] =>
  Array.from({ length: (stop - start) / step + 1 }, (_, i) => ({
    label: (start + i * step).toString(),
    value: (start + i * step).toString(),
  }));
