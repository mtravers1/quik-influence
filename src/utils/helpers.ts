import axios from 'axios';
import { omitBy, isNil } from 'lodash';
import { ADMINS_ID, Q_TOKEN } from './constants';

import { DropdownSelectOption } from 'components/DropdownSelect';
import { FilterDataProps } from 'types';
import { format } from 'date-fns';

export const baseurl = process.env.BACKEND_URL;

export const axiosInstance = axios.create({
  baseURL: `${baseurl}/api/v1`,
  // withCredentials: true,
  headers: {
    'Access-Control-Allow-Headers':
      'Access-Control-Allow-Origin, Access-Control-Allow-Headers, Content-Type',
    'Access-Control-Allow-Origin': '*',
  },
});

export const logout = () => {
  localStorage.removeItem(Q_TOKEN);
  window.location.href = '/login';
};

export const validate = (field: any, pattern: any) => {
  const parts = /\/(.*)\/(.*)/.exec(pattern) || [];
  let restoredRegex = new RegExp(parts[1], parts[2]);
  if (restoredRegex.test(field)) return true;
  return false;
};

export const setToken = (token: string) => {
  axiosInstance.defaults.headers.common['token'] = token;

  if (typeof window !== 'undefined') {
    localStorage.setItem(Q_TOKEN, token);
  }
};

const tokens: any = {};

export function parseJwt(token: any) {
  if (!token) return;
  if (tokens[token]) return tokens[token];

  const base64Url = token.split('.')[1];
  const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
  const jsonPayload = decodeURIComponent(
    atob(base64)
      // Buffer.from(base64, 'base64')
      .split('')
      .map(c => {
        return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
      })
      .join('')
  );

  const result = JSON.parse(jsonPayload);
  tokens[token] = result;

  return result;
}

export function getUser() {
  let user;
  let ctoken;
  let isExpired: boolean = false;

  if (typeof window !== 'undefined') {
    ctoken = localStorage.getItem(Q_TOKEN);
  }

  if (ctoken !== 'null') {
    user = parseJwt(ctoken);
  }

  isExpired = user && user.exp && user.exp < Date.now() / 1000;

  return { admin: user, token: ctoken, isExpired };
}

export const getNumberRange = (
  start: number,
  stop: number,
  step: number
): DropdownSelectOption[] =>
  Array.from({ length: (stop - start) / step + 1 }, (_, i) => ({
    label: (start + i * step).toString(),
    value: (start + i * step).toString(),
  }));

export const getQueryString = (params?: FilterDataProps) => {
  const paramsFilters = omitBy(params, isNil);
  const query = Object.keys(paramsFilters)
    .map(
      k => encodeURIComponent(k) + '=' + encodeURIComponent(paramsFilters[k])
    )
    .join('&');
  return query;
};

export const getSocialHandleHeader = (socialColumns: string[]): string[] => {
  let socialHeader: string[] = [];
  socialColumns.forEach((socialColumn: any) => {
    switch (socialColumn) {
      case 'facebookHandle':
        socialHeader.push('Facebook');
        break;
      case 'twitterHandle':
        socialHeader.push('Twitter');
        break;
      case 'instagramId':
        socialHeader.push('Instagram');
        break;
      case 'tiktokHandle':
        socialHeader.push('Tik Tok');
        break;
      default:
        break;
    }
  });
  return socialHeader;
};

export const similarity = (s1: string, s2: string) => {
  let longer = s1;
  let shorter = s2;
  if (s1.length < s2.length) {
    longer = s2;
    shorter = s1;
  }
  let longerLength = longer.length;
  if (longerLength == 0) {
    return 1.0;
  }
  return (longerLength - editDistance(longer, shorter)) / longerLength;
};

function editDistance(s1: string, s2: string) {
  s1 = s1.toLowerCase();
  s2 = s2.toLowerCase();

  let costs = new Array();
  for (let i = 0; i <= s1.length; i++) {
    let lastValue = i;
    for (let j = 0; j <= s2.length; j++) {
      if (i == 0) costs[j] = j;
      else {
        if (j > 0) {
          let newValue = costs[j - 1];
          if (s1.charAt(i - 1) != s2.charAt(j - 1))
            newValue = Math.min(Math.min(newValue, lastValue), costs[j]) + 1;
          costs[j - 1] = lastValue;
          lastValue = newValue;
        }
      }
    }
    if (i > 0) costs[s2.length] = lastValue;
  }
  return costs[s2.length];
}

export const stringSearch = (val: string, string: string) =>
  string && string.toLowerCase().search(val.toLowerCase()) !== -1;

export function isInViewport(element: any) {
  const rect = element.getBoundingClientRect();
  return (
    rect.top >= 0 &&
    rect.left >= 0 &&
    rect.bottom <=
      (window.innerHeight || document.documentElement.clientHeight) &&
    rect.right <= (window.innerWidth || document.documentElement.clientWidth)
  );
}

export const isAdmin = () => {
  const user = getUser();
  return user.admin && ADMINS_ID.includes(user.admin.roleId);
};

export const hasPermission = (
  userPerm: string[] | undefined,
  permission: string[] | undefined
) => {
  if (userPerm?.length === 0) return true;
  return userPerm?.some(perm => permission?.includes(perm));
};

export const getReportOpenDate = (events: any) => {
  if (!events) return 'NA';
  const openEvent = events.find((evt: any) => evt['event_name'] === 'open');
  if (!openEvent) return 'Not opened yet';

  return format(new Date(openEvent['processed']), 'dd/MM/yyyy');
};

export const getReportUnsub = (events: any) => {
  if (!events) return 'NA';
  const unsubEvent = events.find(
    (evt: any) => evt['event_name'] === 'unsubscribe'
  );
  if (!unsubEvent) return 'No';

  return 'Yes';
};

export const truncateText = (str: string, num: number) => {
  if (str.length <= num) {
    return str;
  }
  return str.slice(0, num) + '...';
};
