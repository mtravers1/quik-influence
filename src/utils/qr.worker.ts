import { QRCodeToDataURLOptions } from 'qrcode';
import { QrCodeRawData } from 'types';

import getQRCodeImages from './getQRCodeImages';

export async function generateImages(
  values: QrCodeRawData[],
  options?: QRCodeToDataURLOptions,
): Promise<string[]> {
  const qrCodeImages = await getQRCodeImages(values, options);
  return qrCodeImages;
}
