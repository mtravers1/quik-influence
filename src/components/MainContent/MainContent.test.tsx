import { renderWithStore } from 'utils/testUtils';
import initialState from '__mockData__/storeData';
import MainContent from '.';

it('renders MainContent component', () => {
  const { container } = renderWithStore(
    <MainContent>
      <>hello world</>
    </MainContent>,
    { initialState }
  );
  expect(container).toMatchSnapshot();
});
