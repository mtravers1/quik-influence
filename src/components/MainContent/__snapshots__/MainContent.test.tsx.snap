// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainContent component 1`] = `
<div>
  <div
    class="chakra-stack css-1a8b1aa"
  >
    <nav
      class="css-1emrlv7"
    >
      <div
        class="css-1c714fi"
      >
        <img
          alt="quik-influence logo"
          class="chakra-image css-1ko1n4n"
        />
        <h1
          class="chakra-heading css-8mmjuy"
        >
          <div
            class="css-k008qs"
          >
            <p
              class="chakra-text css-67ii6d"
            >
              Quik
            </p>
            <p
              class="chakra-text css-0"
            >
              Influence
            </p>
          </div>
        </h1>
      </div>
      <div
        class="css-dsyim5"
      >
        <svg
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="css-0"
      >
        <script>
          (function setScript(initialValue) {
  var mql = window.matchMedia("(prefers-color-scheme: dark)");
  var systemPreference = mql.matches ? "dark" : "light";
  var persistedPreference = systemPreference;

  try {
    persistedPreference = localStorage.getItem("chakra-ui-color-mode");
  } catch (error) {
    console.log("Chakra UI: localStorage is not available. Color mode persistence might not work as expected");
  }

  var colorMode;

  if (persistedPreference) {
    colorMode = persistedPreference;
  } else if (initialValue === "system") {
    colorMode = systemPreference;
  } else {
    colorMode = initialValue != null ? initialValue : systemPreference;
  }

  if (colorMode) {
    /**
     * Keep in sync with \`root.set() {@file ./color-mode.utils.ts}
     */
    document.documentElement.style.setProperty("--chakra-ui-color-mode", colorMode);
    document.documentElement.setAttribute("data-theme", colorMode);
  }
})('dark')
        </script>
      </div>
      <div
        class="css-iqqio3"
      >
        <button
          class="chakra-button css-158uetl"
          type="button"
        >
          Create account
        </button>
        <label
          class="chakra-switch css-tm8ajn"
        >
          <input
            class="chakra-switch__input"
            style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
            type="checkbox"
            value=""
          />
          <span
            aria-hidden="true"
            class="chakra-switch__track css-14qxnv8"
          >
            <span
              class="chakra-switch__thumb css-0"
            />
          </span>
        </label>
        <span
          class="css-1ehzxp"
        >
          <span
            class="chakra-avatar css-o11zrc"
          >
            <div
              aria-label="Segun Adebayo"
              class="chakra-avatar__initials css-0"
              role="img"
            >
              SA
            </div>
          </span>
          <span
            class="css-8uhtka"
          >
            Segun
          </span>
        </span>
      </div>
    </nav>
    <div
      class="css-113h9jy"
    >
      <div
        class="css-1rzqydi"
      >
        <div
          class="css-1h7o23w"
        >
          <a
            class="chakra-link css-1e6bsnr"
            href="/dashboard"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-grip-horizontal "
              data-icon="grip-horizontal"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M96 288H32c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zm160 0h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zm160 0h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zM96 96H32c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zm160 0h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32zm160 0h-64c-17.67 0-32 14.33-32 32v64c0 17.67 14.33 32 32 32h64c17.67 0 32-14.33 32-32v-64c0-17.67-14.33-32-32-32z"
                fill="currentColor"
              />
            </svg>
             
            Overview
          </a>
        </div>
        <div
          class="css-1h7o23w"
        >
          <a
            class="chakra-link css-1e6bsnr"
            href="/campaigns"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-calendar-minus "
              data-icon="calendar-minus"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M436 160H12c-6.6 0-12-5.4-12-12v-36c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48v36c0 6.6-5.4 12-12 12zM12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm304 192c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12H132c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h184z"
                fill="currentColor"
              />
            </svg>
             
            Campaigns
          </a>
        </div>
        <div
          class="css-1h7o23w"
        >
          <a
            class="chakra-link css-1e6bsnr"
            href="/dashboard/credits"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-credit-card "
              data-icon="credit-card"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 432c0 26.5 21.5 48 48 48h480c26.5 0 48-21.5 48-48V256H0v176zm192-68c0-6.6 5.4-12 12-12h136c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H204c-6.6 0-12-5.4-12-12v-40zm-128 0c0-6.6 5.4-12 12-12h72c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM576 80v48H0V80c0-26.5 21.5-48 48-48h480c26.5 0 48 21.5 48 48z"
                fill="currentColor"
              />
            </svg>
             
            Credits
          </a>
        </div>
      </div>
      <hr
        aria-orientation="vertical"
        class="chakra-divider css-78xnth"
      />
      <div
        class="css-1cwxi0e"
      >
        hello world
      </div>
    </div>
  </div>
</div>
`;
